fossid-webapp-client/src/main/kotlin/api/identification/identifiedFiles/IdentifiedFile.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2020 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.fossid.api.identification.identifiedFiles
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.annotation.JsonUnwrapped
+
+import org.ossreviewtoolkit.fossid.api.identification.common.Component
+import org.ossreviewtoolkit.fossid.api.summary.License
+import org.ossreviewtoolkit.fossid.api.summary.Summarizable
+import org.ossreviewtoolkit.fossid.api.summary.SummaryIdentifiedFile
+
+class IdentifiedFile : Summarizable {
+    @JsonProperty("comment")
+    var comment: String? = null
+
+    @JsonProperty("identification_id")
+    var identificationId: Int? = null
+
+    @JsonProperty("identification_copyright")
+    var identificationCopyright: String? = null
+
+    @JsonProperty("is_distributed")
+    var isDistributed: Int? = null
+
+    @JsonProperty("row_id")
+    var rowId: Int? = null
+
+    @JsonProperty("user_name")
+    var userName: String? = null
+
+    @JsonProperty("user_surname")
+    var userSurname: String? = null
+
+    @JsonProperty("user_username")
+    var userUsername: String? = null
+
+    @JsonUnwrapped(prefix = "component_")
+    var component: Component? = null // TODO how to get the additional components ? Only one is returned
+
+    @JsonUnwrapped(prefix = "file_")
+    var file: File? = null
+
+    override fun toSummary(): SummaryIdentifiedFile {
+        val licenses = kotlin.run {
+            if (file!!.licenses == null) {
+                emptyList()
+            } else {
+                file!!.licenses!!.values.map {
+                    License(
+                            identifier = it.identifier!!,
+                            name = it.name,
+                            origin = it.fileLicenseMatchType!!
+                    )
+                }
+            }
+        }
+        return SummaryIdentifiedFile(
+                path = file!!.path!!,
+                comment = comment,
+                licences = licenses)
```

Comment:
Better:

        val licenses = file?.licenses?.let { licenses ->
            licenses.values.map {
                License(
                    identifier = it.identifier!!,
                    name = it.name,
                    origin = it.fileLicenseMatchType!!
                )
            }
        }.orEmpty()

        return SummaryIdentifiedFile(
            path = getFileName(),
            comment = comment,
            licences = licenses
        )

Reply:
But in general, there a way to may non-null assertions in this code for my taste. Are these properties *really* all optional, and you you *really* sure they *always* are non-null in this context? If not, you should probably better choose reasonable default like `identifier = it.identifier.orEmpty()` etc.

Reply:
Fixed for your example.

There are many non-null assertions because all the properties are mutable (`var`).
This is a consequence of using pure Jackson instead of Jackson Kotlin + data classes.

As explained in another answer, we use pure Jackson to leverage the `@JsonUnwrapped` annotation, not supported by Jackson Kotlin.